### ISSUE TITLE
Feature Request: Support for Placeholders for Sagemaker Processing container arguments

### ISSUE THREAD

**Issue by manshahe** [aws/aws-step-functions-data-science-sdk-python#94]
This is a feature request for the newly added **stepfunctions.steps.sagemaker.ProcessingStep** to accept placeholders for inputs to the container_argument parameter. An example case is depicted below:

```
execution_input = ExecutionInput(
    schema={
        "JobName": str,
        #This argument is for the Sagemaker Processing step
        "argument": str
    }
)

processing_step = ProcessingStep(
    job_name=execution_input["JobName"],
    processor={sagemaker processor object here},
    inputs=inputs,
    outputs = outputs,
    #Argument passed here to Sagemaker Processing
    container_arguments=['--argument', execution_input.get['argument'].to_jsonpath()],
    container_entrypoint=["python3", "/opt/ml/processing/input/code/scipt.py"]
)
```

**Rationale:** This will make the generated state machines more generic, we can generate 1 state machine and pass it arguments for different workloads. Without this functionality, I need to generate **_n_** state machines for a parameter which takes **_n_** possible values.

**Comment by wong-a** [comment]
Creating an array with placeholder values for specific elements will require using the `States.Array` [intrinsic function](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-intrinsic-functions.html) unless the referenced JSONPath value is an array.

```json
"AppSpecification.ContainerArguments.$": "States.Array('--argument', $.argument)"
```

**Comment by ca-nguyen** [comment]
The fix to support placeholders (#155) will be available in next release.

**Comment by yanhong-zhao-ef** [comment]
I am wondering if we can re-open this issue since the original feature request is still not possible where execution_input['argument'].to_jsonpath() will get passed straight to the processing script without the variable being replaced on execution and execution_input['argument'] runs into TypeError: Object of type ExecutionInput is not JSON serializable when trying to create workflow
